homeassistant/components/modbus/climate.py

```diff
@@ -108,9 +107,9 @@ def __init__(
         config: dict[str, Any],
     ):
         """Initialize the modbus thermostat."""
-        self._hub: ModbusHub = hub
-        self._name = config[CONF_NAME]
-        self._slave = config.get(CONF_SLAVE)
+        config[CONF_ADDRESS] = "0"
```

Comment:
just wondering, is it safe to mutate the passed in configuration?

Reply:
One caveat I could see is that if the unit test run multiple times against the same configuration object, the next invocation will get the CONF_ADDRESS set to 0

Reply:
That is not totally correct. As long as we run tests against the same configuration object, __init__ will only be called once (when running the setup), and apart from that this is a "dummy" to satisfy the base class, climate do not use CONF_ADDRESS (but will soon).